### ISSUE TITLE
(possibly) Escaping bug of pipes in a table

### ISSUE THREAD

**Issue by bicycle1885** [JuliaDocs/Documenter.jl#148]
I wrote the following table in my Markdown document:
```
| Syntax | Description | Example |
|:-------|:------------|:--------|
| `|` | alternation | `"A|T"` matches `"A"` and `"T"` |
| `*` | zero or more times repeat | `"TA*"` matches `"T"`, `"TA"` and `"TAA"` |
```

But it was rendered like this:
![screen shot 2016-07-27 at 17 21 33](https://cloud.githubusercontent.com/assets/905683/17168669/a25b9652-541e-11e6-90de-d968a0d0a495.png)

I think this is because the converter of Markdown doesn't properly handle pipes (`|`) in a table. The converted Markdown file looks like:
```
Syntax  | Description               | Example                                           
:------ | :------------------------ | :-------------------------------------------------
`       | `                         | alternation                                       
`*`     | zero or more times repeat | `"TA*"` matches `"T"`, `"TA"` and `"TAA"`         
```

Fixing the converted Markdown file by hand solved the problem.

**Comment by MichaelHatherly** [comment]
Recent commit, https://github.com/JuliaLang/julia/commit/15588960c6d44b527156e05edf656d80954b8980, should fix this, kindof.

```
julia> Markdown.parse(
       """
       | Syntax | Description | Example |
       |:-------|:------------|:--------|
       | `\\|` | alternation | `"A\\|T"` matches `"A"` and `"T"` |
       | `*` | zero or more times repeat | `"TA*"` matches `"T"`, `"TA"` and `
       """
       )
Syntax Description               Example                      
–––––– ––––––––––––––––––––––––– –––––––––––––––––––––––––––––
|      alternation               "A|T" matches "A" and "T"    
*      zero or more times repeat "TA*" matches "T", "TA" and `
```

It isn't *really* the nicest fix since it should, in theory, handle `|`s wrapped in backticks fine without needing escaping them, but the way the table parsing is currently done in `Base` makes that pretty difficult to get right. (The whole table parsing method needs a bit of a refactor really, but it's a little late for inclusion in 0.5.)

**Comment by bicycle1885** [comment]
Thank you. I didn't know the backslash escaping have been included in v0.5. That would solve my problem with a small touch-up.

**Comment by MichaelHatherly** [comment]
Closing, since it's fixed upstream already.
